app/models/bookmark.rb

```diff
@@ -2,6 +2,22 @@ class Bookmark < ActiveRecord::Base
   validates :url, presence: true
   belongs_to :topic
   has_many :likes, dependent: :destroy
+  after_validation :set_image
```

Comment:
`after_validation` will trigger the callback after every time the bookmark passes validation. This leaves the bookmark open to fetching new data from the embedly API that is never saved to our db. 

For example `bookmark.valid?` can trigger this callback but will not save any changes made in `set_image` to the db.

You may want a before_save or before_create callback instead here.